specification/dsdl/grammar.tex

```diff
@@ -253,3 +253,15 @@ \subsection{Reserved identifiers}\label{sec:dsdl_reserved_identifiers}
     \texttt{lpt\textbackslash{}d}                      & \texttt{lpt9}      & Compatibility with Microsoft Windows \\
     \texttt{\_.*\_}                                    & \texttt{\_offset\_}& Special-purpose intrinsic entities \\
 \end{CyphalSimpleTable}
+
+\subsection{Reserved comment forms}\label{sec:dsdl_reserved_comment}
+
+Line comments that match the following regular expression are reserved for vendor-specific language extensions:
+\verb|^#\[.+\].*|
```

Comment:
shouldn't this be `^\s*#\s*\[.+\].*`? Either way, it seems like a complex pattern to reserve. What about just `^\s*#!.*` so:

```
# This is a commen[sic]
#! This is an extension. All if it. The entire line!
# Back to business.

```

Reply:
@samcrow could you please provide your input on this?

Reply:
While waiting for Sam to respond, I want to add that this pattern is based on Sam's work on Canadensis, which derives this syntax from Rust, which in turn borrowed it from C#, so we appear to be in good company. There are no strong reasons to maintain consistency with these languages, nor there appear to be reasons to deviate.

Reply:
These are the thoughts I have right now:
* The current ideas use a # followed by a key character, which could be [, !, or something else.
* Allowing whitespace before the # seems like a good idea to be consistent with other languages I've used.
* Allowing whitespace between the # and the key character seems like not a good idea because it would allow breaking up the special sequence of characters.
* The choice of the key character doesn't really matter unless it's something commonly used to start sentences.